pkg/apis/monitoring/v1/types_test.go

```diff
@@ -235,3 +235,75 @@ func TestValidateTLSConfig(t *testing.T) {
 		})
 	}
 }
+
+func TestValidateAuthorization(t *testing.T) {
+	for _, tc := range []struct {
+		config *Authorization
+		err    bool
+	}{
+		{
+			// minimal example
+			config: &Authorization{
+				Credentials: &v1.SecretKeySelector{
```

Comment:
If `Credentials` are always set to the same value, it might be worth extracting the SecretKeySelector into a variable.